
<!DOCTYPE html>


<html lang="en" data-content_root="../../" data-theme="light">

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Example of applied Data Hazards: “Predicting phenotype from genotype” &#8212; Data Hazards v1.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "light";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  <!--
    this give us a css class that will be invisible only if js is disabled
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../../_static/custom.css?v=104d11ec" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=8878045cc6db502f8baf"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf" />

    <script src="../../_static/documentation_options.js?v=e160b93e"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'materials/misc/geno_to_pheno_example';</script>
    <link rel="canonical" href="datahazards.com/materials/misc/geno_to_pheno_example.html" />
    <link rel="icon" href="../../_static/favicon.png"/>
    <link rel="author" title="About these documents" href="../../about.html" />
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="light">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this site..."
         aria-label="Search this site..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
    
    <img src="../../_static/favicon.png" class="logo__image only-light" alt=""/>
    <img src="../../_static/favicon.png" class="logo__image only-dark pst-js-only" alt=""/>
  
  
    <p class="title logo__title">Data Hazards</p>
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../labels.html">
    Data Hazard labels
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../about.html">
    About
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../materials.html">
    How to use
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../contribute.html">
    Contribute
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/very-good-science/data-hazards" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fab fa-github-square fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://osf.io/3fv7t" title="Open Science Framework" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fas fa-book-open fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Open Science Framework</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://twitter.com/hashtag/DataHazards" title="Twitter" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fab fa-twitter-square fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Twitter</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar hide-on-wide">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../labels.html">
    Data Hazard labels
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../about.html">
    About
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../materials.html">
    How to use
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../contribute.html">
    Contribute
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/very-good-science/data-hazards" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fab fa-github-square fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://osf.io/3fv7t" title="Open Science Framework" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fas fa-book-open fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Open Science Framework</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://twitter.com/hashtag/DataHazards" title="Twitter" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fab fa-twitter-square fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Twitter</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
      <div class="sidebar-primary-item">
<div id="ethical-ad-placement"
      class="flat"
      data-ea-publisher="readthedocs"
      data-ea-type="readthedocs-sidebar"
      data-ea-manual="true">
</div></div>
  </div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">

<nav aria-label="Breadcrumb" class="d-print-none">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="../../index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    <li class="breadcrumb-item active" aria-current="page"><span class="ellipsis">Example of applied Data Hazards: “Predicting phenotype from genotype”</span></li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="example-of-applied-data-hazards-predicting-phenotype-from-genotype">
<h1>Example of applied Data Hazards: “Predicting phenotype from genotype”<a class="headerlink" href="#example-of-applied-data-hazards-predicting-phenotype-from-genotype" title="Link to this heading">#</a></h1>
<section id="the-project">
<h2>The project<a class="headerlink" href="#the-project" title="Link to this heading">#</a></h2>
<p>There is so much variation in the human population; from how we look to how we function; from our height to the level of calcium in our blood to our chances of getting Alzheimer’s.
These measurable differences between us are called phenotypes.</p>
<p>Our genotype on the other hand is the variation in our DNA.
This DNA decides exactly which versions of key proteins get made in our bodies, and these proteins go on to have many functions, from being physical building-blocks of cells, to acting as signals to tell the body to produce other proteins.
Sometimes the link between genotype and phenotype are straightforward and known (e.g. for eye colour), but we still do not know even one function for most human proteins.</p>
<p>What we do have is a wealth of information about the smaller building blocks of proteins and links between these and known phenotypes, information about the prevalence of different versions of proteins in the human population, and knowledge about which parts of proteins are conserved across the tree of life (and therefore vital to protein functionality).
This information comes from the literature of links between genes or proteins and function or phenotype across many different organisms, not just humans or even just animals.
We can combine this data and use unsupervised learning (clustering and outlier detection) to create scores predicting which phenotypes each protein effects and predict the likelihood of phenotypes for people within a cohort.
These results could be used to reveal links between genotype and phenotype that could find new treatments for diseases.</p>
<p>The phenotype predictor works by finding people who have an unusual combination of mutations at locations on their DNA that fall in structures with some link to each phenotype.
This cannot make predictions using DNA that falls outside of protein-coding regions, meaning that it uses ~2% of the genome to make these predictions.
It is not expected to be accurate for all phenotypes.
It is not a diagnostic tool, but rather it is supposed to be used as a research tool to find complex traits (which need a combination of mutations) to study further.</p>
<p>The algorithm has been designed and tested for computational efficiency, and is not resource-intensive.</p>
</section>
<section id="how-to-apply-the-hazards">
<h2>How to apply the hazards<a class="headerlink" href="#how-to-apply-the-hazards" title="Link to this heading">#</a></h2>
<p>Go through <a class="reference internal" href="#../../data-hazards.md"><span class="xref myst">the list of hazards</span></a> and try to apply each hazard, looking at the examples and description for clarifications.</p>
</section>
<section id="the-applied-hazards">
<h2>The applied hazards<a class="headerlink" href="#the-applied-hazards" title="Link to this heading">#</a></h2>
<div class="pst-scrollable-table-container"><table class="table">
<tbody>
<tr class="row-odd"><td><p>Hazard Name</p></td>
<td><p>Does the hazard apply?</p></td>
<td><p>Reason</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Contains</span> <span class="pre">Data</span> <span class="pre">Science</span></code></p></td>
<td><p>Yes</p></td>
<td><p>Uses data, makes a prediction, uses unsupervised learning</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">Reinforces</span> <span class="pre">Existing</span> <span class="pre">Biases</span></code></p></td>
<td><p>Yes</p></td>
<td><p>Project does not check that the algorithm works just as well for minority groups, who may be less likely to be represented in the input data. Therefore in a worst-case scenario, this could happen.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Classifies</span> <span class="pre">or</span> <span class="pre">ranks</span> <span class="pre">people</span></code></p></td>
<td><p>Yes</p></td>
<td><p>People are scored (ranked) for different phenotypes, including phenotypes relating to mental health and disability.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">High</span> <span class="pre">Environmental</span> <span class="pre">Cost</span></code></p></td>
<td><p>No</p></td>
<td><p>The algorithm has been tested for computationally efficiency and does not require many computational resources.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Lacks</span> <span class="pre">Community</span> <span class="pre">Involvement</span></code></p></td>
<td><p>Yes</p></td>
<td><p>The communities of people with the phenotypes have no current involvement in this process.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">Danger</span> <span class="pre">of</span> <span class="pre">misuse</span></code></p></td>
<td><p>Yes</p></td>
<td><p>The phenotype predictor is not expected to be accurate for all phenotypes, but It could even be used to try to predict phenotypes that are caused by the environment or regions of DNA it does not consider, if these are defined as genetic phenotypes in other literature.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Difficult</span> <span class="pre">to</span> <span class="pre">understand</span></code></p></td>
<td><p>Yes</p></td>
<td><p>Doesn’t use black box machine learning (e.g. deep learning), but has closed source code and a complicated data pipeline.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">May</span> <span class="pre">cause</span> <span class="pre">direct</span> <span class="pre">harm</span></code></p></td>
<td><p>No</p></td>
<td><p>Could not cause direct harm unless misused.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Privacy</span></code></p></td>
<td><p>Yes</p></td>
<td><p>Individual’s genetic data is required to run the phenotype predictor. This has many privacy risks, for example identification, use by insurers, being contacted by unknown relatives.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">Automates</span> <span class="pre">decision-making</span></code></p></td>
<td><p>No</p></td>
<td><p>The phenotype predictor is not a diagnostic tool.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">Lacks</span> <span class="pre">informed</span> <span class="pre">consent</span></code></p></td>
<td><p>No</p></td>
<td><p>The predictor has used datasets from 1000 genomes project, ALSPAC, and genetrainer. Although these individuals were not specifically aware that their data would be used in this way, they were made aware that their data could be used for a range of research.</p></td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="applying-the-safety-precautions">
<h2>Applying the safety precautions<a class="headerlink" href="#applying-the-safety-precautions" title="Link to this heading">#</a></h2>
<p>The applied hazards can then be used to help the researcher create a list of safety precautions for their project using those provided on the <a class="reference internal" href="#../../data-hazards.md"><span class="xref myst">hazards page</span></a>.
In this case the safety precautions might be:</p>
<ul class="simple">
<li><p>Clearly lay out responsibilities for research and adoption/deployment.</p></li>
<li><p>Test the algorithm’s efficacy separately for minority groups.</p></li>
<li><p>Curate the phenotypes predicted on by the released software. In deployment: ensure this information is only shared with users themselves, and that they have assistance interpreting the scores.</p></li>
<li><p>Phenotypes chosen as case-studies for research should be chosen carefully and involve communities where relevant. For it to be deployed, the list of conditions that it predicts should be curated manually depending on the community. This would prevent situations, such as the algorithm being used to predict whether unborn children are likely to be neurodivergent.</p></li>
<li><p>State the limitations of the method clearly, retest it if it is used outside the research context.</p></li>
<li><p>Increase documentation, and make the source code open if possible.</p></li>
<li><p>Informed consent of any additional users/participants, relating to privacy as well as other aspects. Store data carefully/briefly.</p></li>
</ul>
</section>
</section>


                </article>
              
              
              
              
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
<div
    id="pst-page-navigation-heading-2"
    class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc" aria-labelledby="pst-page-navigation-heading-2">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-project">The project</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#how-to-apply-the-hazards">How to apply the hazards</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-applied-hazards">The applied hazards</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#applying-the-safety-precautions">Applying the safety precautions</a></li>
</ul>
  </nav></div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/very-good-science/data-hazards/edit/main/site/materials/misc/geno_to_pheno_example.md">
      <i class="fa-solid fa-pencil"></i>
      
      
        
          Edit on GitHub
        
      
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item">
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../../_sources/materials/misc/geno_to_pheno_example.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2024, Natalie Zelenka and Nina Di Cara.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item">

  <p class="sphinx-version">
    Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 7.4.7.
    <br/>
  </p>
</div>
      
    </div>
  
  
  
    <div class="footer-items__end">
      
        <div class="footer-item">
<p class="theme-version">
  <!-- # L10n: Setting the PST URL as an argument as this does not need to be localized -->
  Built with the <a href="https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html">PyData Sphinx Theme</a> 0.16.1.
</p></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>